### ISSUE TITLE
Don't complain about methods where the arity is overloaded

### ISSUE THREAD

**Issue by backus** [backus/yardcheck#10]
We should not flag this as wrong. Note that it documents both `percent_literal?` and `percent_literal?(type)`

```
Expected RuboCop::AST::ArrayNode#percent_literal? to receive Symbol for type but observed NilClass

    source: lib/rubocop/ast/node/array_node.rb:36
    tests:
      - ./spec/rubocop/ast/array_node_spec.rb:52
      - ./spec/rubocop/ast/array_node_spec.rb:60
      - ./spec/rubocop/ast/array_node_spec.rb:68

    # Checks whether the `array` literal is delimited by percent brackets.
    #
    # @overload percent_literal?
    #   Check for any percent literal.
    # @overload percent_literal?(type)
    #   Check for percent literaly of type `type`.
    # @param type [Symbol] an optional percent literal type
    # @return [Boolean] whether the array is enclosed in percent brackets
    def percent_literal?(type = nil)
      if type
        loc.begin && loc.begin.source =~ PERCENT_LITERAL_TYPES[type]
      else
        loc.begin && loc.begin.source.start_with?('%')
      end
    end
```

**Comment by backus** [comment]
Actually this isn't a bug if you use `@overload` properly! That method should have been documented like this:

```ruby
# Checks whether the `array` literal is delimited by percent brackets.
#
# @overload percent_literal?
#   Check for any percent literal.
#
# @overload percent_literal?(type)
#   Check for percent literaly of type `type`.
#
#   @param type [Symbol] an optional percent literal type
#
# @return [Boolean] whether the array is enclosed in percent brackets
def percent_literal?(type = nil)
  if type
    loc.begin && loc.begin.source =~ PERCENT_LITERAL_TYPES[type]
  else
    loc.begin && loc.begin.source.start_with?('%')
  end
end
```

**Comment by backus** [comment]
Ah this is because there aren't any directly accessible param tags when overloaded:

```
>> YARD::Registry.all(:method).select { |m| m.to_s =~ /percent_literal\?/ }.first.tags
=> [
  #<YARD::Tags::OverloadTag:0x007fc8d528a190 @tag_name="overload", @text=nil, @name=:percent_literal?, @types=nil, @signature="percent_literal?", @docstring="Check for any percent literal.", @parameters=[], @object=#<yardoc method RuboCop::AST::ArrayNode#percent_literal?>>,
  #<YARD::Tags::OverloadTag:0x007fc8d528a050 @tag_name="overload", @text=nil, @name=:percent_literal?, @types=nil, @signature="percent_literal?(type)", @docstring="Check for percent literaly of type `type`.", @parameters=[["type", nil]], @object=#<yardoc method RuboCop::AST::ArrayNode#percent_literal?>>,
  #<YARD::Tags::Tag:0x007fc8d5289dd0 @tag_name="return", @text="whether the array is enclosed in percent brackets", @name=nil, @types=["Boolean"], @object=#<yardoc method RuboCop::AST::ArrayNode#percent_literal?>>
]
```
